gadgets/trace_tcpdrop/program.bpf.c

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Based on tcpdrop(8) from BCC
+//
+// Copyright 2018 Netflix, Inc.
+// 30-May-2018    Brendan Gregg   Created this.
+// 15-Jun-2022    Rong Tao        Add tracepoint:skb:kfree_skb
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include <gadget/buffer.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+
+#define GADGET_TYPE_TRACING
+#include <gadget/sockets-map.h>
+
+#define TASK_COMM_LEN 16
+
+struct event {
+	struct gadget_l4endpoint_t src;
+	struct gadget_l4endpoint_t dst;
+
+	gadget_timestamp timestamp;
+	__u8 state;
+	__u8 tcpflags;
```

Comment:
I think we don't have support for bitfields yet
https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/design/002-containerized-gadgets.md#bitfield

We should file an issue for that. It should be similar to the issue for enums (#1928).

Definition in kernel:
https://github.com/torvalds/linux/blob/v6.2/include/net/tcp.h#L840-L847

This can be addressed in a future PR.

Reply:
Opened https://github.com/inspektor-gadget/inspektor-gadget/issues/2482